rust/sdk-processor/src/steps/parquet_default_processor/gcs_handler.rs

```diff
@@ -0,0 +1,185 @@
+use crate::parquet_processors::{ParquetTypeEnum, ParquetTypeStructs};
+use anyhow::Context;
+use aptos_indexer_processor_sdk::utils::errors::ProcessorError;
+use async_trait::async_trait;
+use google_cloud_storage::client::Client as GCSClient;
+use parquet::{
+    file::{properties::WriterProperties, writer::SerializedFileWriter},
+    record::RecordWriter,
+    schema::types::Type,
+};
+use processor::bq_analytics::{
+    gcs_handler::upload_parquet_to_gcs,
+    generic_parquet_processor::{GetTimeStamp, HasParquetSchema, HasVersion},
+};
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+use tracing::{debug, error};
+
+pub struct GCSUploader {
+    gcs_client: Arc<GCSClient>,
+    parquet_type_to_schemas: HashMap<ParquetTypeEnum, Arc<Type>>,
+    parquet_type_to_writer: HashMap<ParquetTypeEnum, SerializedFileWriter<Vec<u8>>>,
+    pub bucket_name: String,
+    pub bucket_root: String,
+    pub processor_name: String,
+}
+
+#[async_trait]
+pub trait Uploadable {
+    async fn handle_buffer(
```

Comment:
Should this be called `upload_buffer`? `handle` is very generic-sounding

Reply:
yeah, upload makes more sense!

Reply:
Doesn't look like this was addressed so opening back up!

Reply:
oops I think Imissed, updated!